tools/flaky-test-reporter/json_reporter.go

```diff
@@ -24,6 +24,14 @@ import (
 	"knative.dev/test-infra/tools/flaky-test-reporter/jsonreport"
 )
 
+type ReportClient struct {
```

Comment:
why do we need another client? Can this be part of JSONClient itself?

Reply:
as with log_parser.go, I am following a similar pattern to ghutil and its uses in the reporter and retryer. I feel like moving these methods to jsonreport would make it too specific to the reporter itself.